azurelinuxagent/ga/collect_logs.py

```diff
@@ -85,10 +84,7 @@ def __init__(self):
         self.event_thread = None
         self.should_run = True
         self.last_state = None
-
```

Comment:
There was no need to use PeriodicOperations in the log collector thread. This was probably copy and pasted from the Monitor thread.

Reply:
The thread blocks waiting for the log collector. If we did this on one of the other threads we may end up blocking all the operations in that thread while the log collector runs. With a little more of extra code we could just start the collector and not block on it, similarly to what we do when we start extensions. I think the code was written like this to make it a little simpler. We can always improve on it later.